src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/EventDateAndTimeControllerTests.cs

```diff
@@ -139,6 +163,59 @@ public void Post_HasSeenPreview_True_RedirectsToCheckYourAnswers()
         result.RouteName.Should().Be(RouteNames.CreateEvent.CheckYourAnswers);
     }
 
+    [Test, MoqAutoData]
+    public void Get_IsAlreadyPublishedTrue_PostLinkLinkIsUpdateCalendarEvent()
+    {
+        var sessionServiceMock = new Mock<ISessionService>();
+        var validatorMock = new Mock<IValidator<EventDateAndTimeViewModel>>();
+
+        var sessionModel = new EventSessionModel { IsAlreadyPublished = true };
+
+        sessionServiceMock.Setup(s => s.Get<EventSessionModel>()).Returns(sessionModel);
+
+        var sut = new EventDateAndTimeController(sessionServiceMock.Object, validatorMock.Object);
+
+
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.UpdateEvent.UpdateDateAndTime, UpdateEventDateAndTimeUrl);
+        var result = (ViewResult)sut.Get();
+
+        Assert.That(result.Model, Is.TypeOf<EventDateAndTimeViewModel>());
+        var vm = result.Model as EventDateAndTimeViewModel;
+        vm!.PostLink.Should().Be(UpdateEventDateAndTimeUrl);
+    }
+
+    [Test, MoqAutoData]
+    public void Post_IsAlreadyPublishedTrue_RedirectsToUpdateDateAndTime()
```

Comment:
The test name is incorrect I think as the redirection is to CalendarEvent?

Reply:
ok